exp/tools/serpapi/internal/serpapiClient.go

```diff
@@ -0,0 +1,108 @@
+package internal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+var ErrMissingToken = errors.New("missing the OpenAI API key, set it in the OPENAI_API_KEY environment variable")
+var NoGoodResult = "No good search result found"
+
+type SerpapiClient struct {
+	apiKey  string
+	baseURL string
+}
+
+func New() (*SerpapiClient, error) {
+	apiKey := os.Getenv("SERPAPI_API_KEY")
+	if apiKey == "" {
+		return nil, ErrMissingToken
+	}
+
+	return &SerpapiClient{
+			apiKey:  apiKey,
+			baseURL: "https://serpapi.com/search",
+		},
+		nil
+}
+
+func (s *SerpapiClient) Search(query string) (string, error) {
+	var params url.Values
+	query = strings.ReplaceAll(query, " ", "+")
+	params.Set("q", query)
+	params.Add("google_domain", "google.com")
+	params.Add("gl", "us")
+	params.Add("hl", "en")
+	params.Add("api_key", s.apiKey)
+
+	reqURL := fmt.Sprintf("%s?%s", s.baseURL, params.Encode())
+	resp, err := http.Get(reqURL)
+	if err != nil {
+		fmt.Printf("Error: %s\n", err)
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Printf("Error: %s\n", err)
+		return "", err
+	}
+	var result map[string]interface{}
+	err = json.Unmarshal([]byte(body), &result)
+	if err != nil {
+		fmt.Printf("Error: %s\n", err)
```

Comment:
Please do not use `fmt.Printf(...)` to log out the error/warn messages, please use the `log` standard logging package to log messages with prefixes.

Btw, I do know there are some bad usage of it under the `exp` sub-directory, we could fix them later.

@tmc how about later create a logger package along with methods and ask the other packages to use it globally instead of using `fmt` and `log` separately which could cause difficulty to refactor the codes?